benchmark_v2/reformat_benchmark_results.py

```diff
@@ -0,0 +1,415 @@
+    # "str-groupby",
+    # "str-gather",
+    # "str-in1d",
+    # "substring_search",
+    # "split",
+    # "sort-cases",
+    # "multiIO",
+    # "str-locality",
+    # "dataframe",
+    # "encode",
+    # "bigint_conversion",
+    # "bigint_stream",
+    # "bigint_bitwise_binops",
+    # "bigint_groupby",
+    # "bigint_array_transfer",
+]
+
+if os.getenv("ARKOUDA_SERVER_PARQUET_SUPPORT"):
+    BENCHMARKS.append("parquetIO")
+    BENCHMARKS.append("parquetMultiIO")
+
+
+def get_chpl_util_dir():
+    """Get the Chapel directory that contains graph generation utilities."""
+    CHPL_HOME = os.getenv("CHPL_HOME")
+    if not CHPL_HOME:
+        logging.error("$CHPL_HOME not set")
+        sys.exit(1)
+    chpl_util_dir = os.path.join(CHPL_HOME, "util", "test")
+    if not os.path.isdir(chpl_util_dir):
+        logging.error("{} does not exist".format(chpl_util_dir))
+        sys.exit(1)
+    return chpl_util_dir
+
+
+def generate_graphs(args):
+    """
+    Generate graphs using the existing .dat files and graph infrastructure.
+    """
+    genGraphs = os.path.join(get_chpl_util_dir(), "genGraphs")
+    cmd = [
+        genGraphs,
+        "--perfdir",
+        args.dat_dir,
+        "--outdir",
+        args.graph_dir,
+        "--graphlist",
+        os.path.join(args.graph_infra, "GRAPHLIST"),
+        "--testdir",
+        args.graph_infra,
+        "--alttitle",
+        "Arkouda Performance Graphs",
+    ]
+
+    if args.platform_name:
+        cmd += ["--name", args.platform_name]
+    if args.configs:
+        cmd += ["--configs", args.configs]
+    if args.start_date:
+        cmd += ["--startdate", args.start_date]
+    if args.annotations:
+        cmd += ["--annotate", args.annotations]
+
+    subprocess.check_output(cmd)
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument(
+        "--dat-dir",
+        default=os.path.join(benchmark_dir, "datdir"),
+        help="Directory with .dat files stored",
+    )
+    parser.add_argument("--graph-dir", help="Directory to place generated graphs")
+    parser.add_argument(
+        "--graph-infra",
+        default=os.path.join(benchmark_dir, "graph_infra"),
+        help="Directory containing graph infrastructure",
+    )
+    parser.add_argument("--platform-name", default="", help="Test platform name")
+    parser.add_argument("--description", default="", help="Description of this configuration")
+    parser.add_argument("--annotations", default="", help="File containing annotations")
+    parser.add_argument("--configs", help="comma seperate list of configurations")
+    parser.add_argument("--start-date", help="graph start date")
+    parser.add_argument("--benchmark-data", help="the benchnmark output data in json format.")
+
+    return parser
+
+
+def get_header_dict(directory_path):
+    """Get a dictionary there the keys are the benchmark name and the values are list of header fields for the .dat file."""
```

Comment:
This docstring appears below also, and looks like it was probably cut-and-pasted from somewhere.  It's just a bit unclear and ungrammatical.  Not a show-stopper, but could you look at this and maybe improve it?